lib/pc_patch.c

```diff
@@ -544,3 +544,60 @@ PCPOINT *pc_patch_pointn(const PCPATCH *patch, int n)
 	pcerror("%s: unsupported compression %d requested", __func__, patch->type);
 	return NULL;
 }
+
+/** set schema for patch */
+PCPATCH*
+pc_patch_set_schema(const PCPATCH *patch, const PCSCHEMA *new_schema)
+{
+	PCPATCH_UNCOMPRESSED *paout = NULL;
+	PCPOINTLIST *opl, *npl;
+	PCPOINT *opt, *npt;
+	double val;
+	size_t i, j;
+	char *name;
+	PCSCHEMA *old_schema;
+
+	old_schema = (PCSCHEMA *) patch->schema;
+
+	if ( pc_schema_similar(old_schema, new_schema) )
+	{
+		// fast path!
+		old_schema->pcid = new_schema->pcid;
+		return (PCPATCH *) patch;
```

Comment:
Why not `patch->schema=new_schema` so that we can garantee that the returned patch has the new schema, rather than only a similar one ?